### PR TITLE
 🏃 Run tests with race detector enabled

### PR DESCRIPTION
**Pull request by alvaroaleman**
Changes the test script to run with `--race` and fixes the ~three~ five races found (all of which are in test code)

### CODE REVIEW

File: pkg/manager/manager_test.go [hunk 1]
```diff
@@ -845,11 +850,13 @@ var _ = Describe("manger.Manager", func() {
 		m, err := New(cfg, Options{})
 		Expect(err).NotTo(HaveOccurred())
 		startGoruntime := rt.NumGoroutine()
+		_ = pprof.Lookup("goroutine").WriteTo(os.Stdout, 1)
```
Comment (alvaroaleman) [inline pkg/manager/manager_test.go:853]: The test here is apparently flaky with `-race` set. I am not sure if we can fix this, but having the dump here will at least give some clues.

I am unfortunately not able to repro the issue on my computer, that's why I left this here.

**Comment by alvaroaleman** [comment]
Okay, that failure was presumably cpu starvation:
```
"timeout waiting for process kube-apiserver to start
```

Generally running the tests with `-race` means they need much more cpu so we might have to bump the requests of the job (But I think using the race detector provides a lot of value)
/retest

**Comment by vincepri** [comment]
We could limit the amount of parallel tests with `-p`

**Comment by alvaroaleman** [comment]
> We could limit the amount of parallel tests with -p

Do you know how that interacts with ginkgo?

**Comment by vincepri** [comment]
Ah, forgot about we were using ginkgo — I'll need to look into that, we can set `GINKGO_NODES`, which IIRC defaults to 1

**Comment by alvaroaleman** [comment]
@vincepri wdyt about merging this? If we end up getting flakes we could for example use build flags to run a subset of tests without race detector

**Pull request merged**